File: codes/lua/operator_ternary.lua
print('\n-- Ternary Operator in Lua')

function ternary_operator_v1(true_condition, value_if_condition_is_true, value_if_condition_is_false)
    if true_condition then
        return value_if_condition_is_true
    end
    return value_if_condition_is_false
end

function ternary_operator_v2(true_condition, value_if_condition_is_true, value_if_condition_is_false)
    return ((true_condition and value_if_condition_is_true) or value_if_condition_is_false)
end

CORRECT_ANSWER = "foo"
print("correct answer: " .. CORRECT_ANSWER)

print("-- using Ternary Operator function \"ternary_operator_v1\"")

my_answer = "bar"
print("my answer: " .. my_answer)
print("is my answer correct: " .. tostring(ternary_operator_v1((my_answer == CORRECT_ANSWER), "correct!", "wrong!")))
-- is my answer correct: wrong!

my_answer = "foo"
print("my answer: " .. my_answer)
print("is my answer correct: " .. tostring(ternary_operator_v1((my_answer == CORRECT_ANSWER), "correct!", "wrong!")))
-- is my answer correct: correct!

print("-- using Ternary Operator function \"ternary_operator_v2\"")

my_answer = "bar"
print("my answer: " .. my_answer)
print("is my answer correct: " .. tostring(ternary_operator_v2((my_answer == CORRECT_ANSWER), "correct!", "wrong!")))
-- is my answer correct: wrong!

my_answer = "foo"
print("my answer: " .. my_answer)
print("is my answer correct: " .. tostring(ternary_operator_v2((my_answer == CORRECT_ANSWER), "correct!", "wrong!")))
-- is my answer correct: correct!

my_answer = "bar"
print("my answer: " .. my_answer)
print("is my answer correct: " .. tostring(((my_answer == CORRECT_ANSWER) and "correct!") or "wrong!"))
-- is my answer correct: wrong!

my_answer = "foo"
print("my answer: " .. my_answer)
print("is my answer correct: " .. tostring(((my_answer == CORRECT_ANSWER) and "correct!") or "wrong!"))
-- is my answer correct: correct!

File: codes/kotlin/operator_ternary.kt
@Suppress("UNUSED_VARIABLE", "UNCHECKED_CAST")
// to compile and run the code:
// kotlinc operator_ternary.kt -include-runtime -d operator_ternary.jar && kotlin operator_ternary.jar

fun main() {
    println("\n// Ternary Operator in Kotlin")

    val CORRECT_ANSWER = "foo"
    println("correct answer: ${CORRECT_ANSWER}")

    var myAnswer: Any?

    myAnswer = "bar"
    println("my answer: ${myAnswer}")
    println("is my answer correct: ${if (myAnswer == CORRECT_ANSWER) "correct!" else "wrong!"}")
    // is my answer correct: wrong!

    myAnswer = "foo"
    println("my answer: ${myAnswer}")
    println("is my answer correct: ${if (myAnswer == CORRECT_ANSWER) "correct!" else "wrong!"}")
    // is my answer correct: correct!
}
